security/authorizationreloader.go

```diff
@@ -0,0 +1,73 @@
+package security
+
+import (
+	"context"
+
+	commonservice "github.com/cloudtrust/common-service/v2"
+	"github.com/cloudtrust/common-service/v2/idgenerator"
+	"github.com/cloudtrust/common-service/v2/log"
+	kafkauniverse "github.com/cloudtrust/kafka-client"
+	"github.com/spf13/viper"
+)
+
+type authorizationReloader struct {
+	authManager     AuthorizationManager
+	kafkaUniverse   *kafkauniverse.KafkaUniverse
+	kafkaConsumerID string
+}
+
+type AuthorizationRelaoder interface {
+	StartConsumers()
+	Close()
+}
+
+func NewAuthorizationReloader(id string, authManager AuthorizationManager, logger log.Logger, ctx context.Context, v *viper.Viper, idGenerator idgenerator.IDGenerator, logEventRate int64) AuthorizationRelaoder {
```

Comment:
Would it be possible not to link this module with kafka-client?
We don't want to have multiple "KafkaUniverse" instances in a single application. May be we could have a way to link a given consumer of the KafkaUniverse to a callback function (something like func OnReceivedContent(byte[] content) error).
This callback function would be the entry point in the AuthorizationManager. This way, we don't need an authorization reloader, just a callback in AuthorizationManager and have no link with Kafka

kafkaUniverse.GetConsumer(id).
		...
		SetHandler(authManager.OnReceivedContent)

Reply:
Okay so if I understand correctly you suggest to only modify the kafka-client not the common-service.

Reply:
I suggest to modify:
- the kafka-client library to add an easy way of launching KafkaUniverse when you only need to run one consumer providing a basic consumer...
- the common-service library to be able to update the AuthorizationManager providing a simple input (probably of type []byte)

Reply:
I agree with Francis !